core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1498,7 +1498,7 @@ public static SqlNode toSql(RexLiteral literal) {
     case EXACT_NUMERIC: {
       if (SqlTypeName.APPROX_TYPES.contains(typeName)) {
         return SqlLiteral.createApproxNumeric(
-            castNonNull(literal.getValueAs(BigDecimal.class)).toPlainString(), POS);
+            castNonNull(literal.getValueAs(Double.class)).toString(), POS);
```

Comment:
Maybe we should add Javadoc to prevent someone from changing it back to BigDecimal later.

Reply:
If they change this to BigDecimal the program will crash with a type cast error.

Reply:
It might crash today, but it's possible that someone will enhance `RexLiteral.getValueAs` to accept `BigDecimal.class` as an argument. That's not an unreasonable change.

I had different concerns about this code, namely that assumes that how Java formats Double values is the same as how SQL dialects represent FLOAT literals.

I am satisfied that code is OK only because there are already quite extensive unit tests for SQL generation.

Reply:
BigDecimal.class should still be a legal argument for this function. 
There are a few places in the code where this happens.

This code assumes only that the createApproxNumeric will correctly handle Java string representations of double values.